File: SimpleHabit/app/src/main/java/com/me/simplehabit/persistence/dao/CurrentProgramDao.java
package com.me.simplehabit.persistence.dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import com.me.simplehabit.data.vos.CurrentProgramVO;

@Dao
public interface CurrentProgramDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void saveCurrentProgram(CurrentProgramVO currentProgram);

    @Query("select * from CurrentProgram")
    CurrentProgramVO retrieveCurrentProgram();
}

File: SimpleHabit/app/src/main/java/com/me/simplehabit/persistence/dao/CurrentProgramDao.kt
package com.me.simplehabit.persistence.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

import com.me.simplehabit.data.vos.CurrentProgramVO

@Dao
interface CurrentProgramDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun saveCurrentProgram(currentProgram: CurrentProgramVO)

    @Query("select * from CurrentProgram")
    fun retrieveCurrentProgram(): CurrentProgramVO

    @Query("select count(*) from CurrentProgram")
    fun retrieveCurrentProgramCount(): Int
}
